### DISCUSSION TITLE
[PSD16] Esercizio Slide 17

### DISCUSSION THREAD

**Discussion by matypist** [sapienzastudentsnetwork/psd2223#218] [Esercizi (A-L)]
![image](https://user-images.githubusercontent.com/69389461/215794245-1f2cf258-acfe-4744-bac3-c1f3c37ec9a8.png)

**Comment by Elia-Belli** [comment]
### Frequenza Massima
La frequenza è l'inverso del periodo (del clock), la disciplina temporale ci impone alcuni vincoli sul periodo minimo del clock $T_c$ affinchè il circuito operi in modo affidabile, ed applicando i vari vincoli si arriva alla disequazione: 
$$T_c \geq t_{pcq} + t_{pd} + t_{su}$$
Ogni volta che si calcola un delay si valuta sempre il caso peggiore, ovvero il **critical path**, quindi in questo caso $t_{pd} = 3 \times t_{gate\ pd}$ (anche se le porte sono diverse l'esercizio assegna a tutte lo stesso delay), di conseguenza:
$T_c \geq 200ps \rightarrow F_{max} = 1/200\ ps^{-1} = 5\ GHz$
### Latenza e Portata
La latenza $L$ è il tempo impiegato ad elaborare un token (un input o insieme di input), ovvero il tempo che passa dall'entrata del token nel circuito alla sua uscita, in questo caso $L = T_c = 200\ ps$ ed essendo la portata l'inverso della latenza, essa è pari a $5\times 10^9\ token/s$.
